controller/auth-controller.js

```diff
@@ -0,0 +1,103 @@
+const bcrypt = require("bcryptjs");
+const jwt = require("jsonwebtoken");
+
+const User = require("../models/user");
+
+exports.login = async (req, res) => {
+  const {name, password} = req.body
+  const key = process.env.JWT_SECRET;
+
+  const user = await User.findOne({ userName: name });
+
+  if (!user) {
+    res.json({
+      err: "User dosent exist",
+    });
+    return;
+  }
+
+  const isAuth = await bcrypt.compare(password, user.password);
+  if (isAuth) {
+    const payLoad = {
+      name: user.userName,
+      password: user.password,
+    };
+
+    const token = await jwt.sign(payLoad, key);
+
+    res.json({
+      success: "logged in",
+      token
+    });
+  } else {
+    res.json({
+      err: "Wrong Password",
+    });
+  }
+};
+
+exports.signUp = async (req, res) => {
+  const {name, password} = req.body
+
+  const user = await User.findOne({ userName: name });
+  if (user) {
+    res.json({
+      err: "User already exists",
+    });
+    return;
+  }
+
+  const hashedPassword = await bcrypt.hash(password, 10);
+
+  let newUser = new User({
+    userName: name,
+    password: hashedPassword,
+  });
+
+  newUSer = await newUser.save((err) => {
+    if (err) {
+      res.json({
+        err,
+      });
+      return;
+    }
+  });
+
+  res.json({
+    success: "successfully signed up",
+    id:newUser._id
+  });
+};
+
+exports.isAuth = (req, res, next) => {
+  if (!req.headers.auth) {
+    res.json({
+      err: "not logged in",
+    });
+    return;
+  }
+
+  let token = req.headers.auth;
```

Comment:
Assign this at the top of function body?

Reply:
done